### PR TITLE
Build attributes from glib properties whenever possible

### PR DESCRIPTION
**Pull request by k0ral**
It makes it possible to use `notifyProperty` (from [here][1]).
The code depends on https://github.com/gtk2hs/gtk2hs/pull/94 .

[1]: https://hackage.haskell.org/package/gtk3-0.13.4/docs/Graphics-UI-Gtk-Abstract-Object.html

### CODE REVIEW

**Comment by k0ral** [comment]
I've updated the dependency on glib, now that gtk2hs/gtk2hs#94 is merged.

**Comment by dmwit** [comment]
It looks like webkit doesn't build with these changes. (Perhaps some things should only be exposed from webkitgtk3.) Could you take a look at that?

**Comment by k0ral** [comment]
The build only failed for GHC 7.4, while it succeeded for 7.6 and 7.8 .
The error doesn't seem related to my pull request:

```
cairo-0.13.1.0 failed during the configure step. The exception was:
user error
(/opt/ghc/7.4.2/lib/ghc-7.4.2/unix-2.5.1.1/libHSunix-2.5.1.1.a(execvpe.o): In
function `pPrPr_disableITimers':
(.text+0x320): multiple definition of `pPrPr_disableITimers'
```

**Comment by k0ral** [comment]
Builds for GHC 7.4 have been broken since 4a2726f , so I don't believe this PR has anything to do with it.

**Comment by dmwit** [comment]
Commit de5e355 (as of time of writing, that is master here) builds fine for me on 7.8.4. Your changes result in this error (again on 7.8.4):

    Graphics/UI/Gtk/WebKit/WebView.chs:805:19:
        Not in scope: ‘readAttrFromDoubleProperty’
        Perhaps you meant one of these:
          ‘newAttrFromDoubleProperty’ (imported from System.Glib.Properties),
          ‘readAttrFromBoolProperty’ (imported from System.Glib.Properties),
          ‘readAttrFromEnumProperty’ (imported from System.Glib.Properties)

Are you running `install-both.sh` to make sure you install both webkit *and* webkitgtk3 (or putting the appropriate cabal file in place to build only webkit)?

**Comment by k0ral** [comment]
I had forgotten to bump the version for glib's dependency (`readAttrFromDoubleProperty` was introduced in 0.13.1.0). This is fixed now.

File: Graphics/UI/Gtk/WebKit/WebView.chs [hunk 1]
```diff
@@ -1,4 +1,5 @@
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE CPP          #-}
+{-# LANGUAGE ViewPatterns #-}
```
Comment (hamishmack) [inline Graphics/UI/Gtk/WebKit/WebView.chs]: Are ViewPatterns used?

File: Graphics/UI/Gtk/WebKit/WebView.chs [hunk 2]
```diff
@@ -767,37 +774,35 @@ webViewGetPasteTargetList = maybePeek mkTargetList <=< {#call web_view_get_paste
 
 -- * Attibutes
 
+#if WEBKIT_CHECK_VERSION(1,0,1)
 -- | Zoom level of the 'WebView' instance
 webViewZoomLevel :: WebViewClass self => Attr self Float
-webViewZoomLevel = newAttr
-   webViewGetZoomLevel
-   webViewSetZoomLevel
+webViewZoomLevel = newAttrFromFloatProperty "zoom-level"
```
Comment (hamishmack) [inline Graphics/UI/Gtk/WebKit/WebView.chs]: Looking in the webkit code it looks like webkit_web_get/set_view_zoom_level (the C functions used by webViewGet/SetZoomLevel) are much more direct.  Are we sure we want to introduce the overhead of looking up the property by name?

File: Graphics/UI/Gtk/WebKit/WebView.chs (hunk 1 above)
Comment (k0ral) [inline Graphics/UI/Gtk/WebKit/WebView.chs]: Yes, cf line 486 . I can remove its use if you consider it edgy.

Comment (hamishmack) [inline Graphics/UI/Gtk/WebKit/WebView.chs]: oh yeah.  I'm fine with it.

File: Graphics/UI/Gtk/WebKit/WebView.chs (hunk 2 above)
Comment (k0ral) [inline Graphics/UI/Gtk/WebKit/WebView.chs]: The problem with using the direct getter/setter is that it generates an [attribute][1] that is not a [property] (in the glib's sense). And only properties can trigger notifications through `Graphics.UI.Gtk.Abstract.Object.notifyProperty`, so it looks like the overhead is necessary to get this feature.

[1]: https://hackage.haskell.org/package/glib-0.13.1.0/docs/System-Glib-Attributes.html
[2]: https://hackage.haskell.org/package/glib-0.13.1.0/docs/System-Glib-Properties.html

Comment (hamishmack) [inline Graphics/UI/Gtk/WebKit/WebView.chs]: Ah!

**Pull request merged**